pkg/api/healthport/healthport.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package healthport
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+)
+
+const defaultTimeout = time.Second
+
+var server *http.Server
+
+// Serve configures and starts the http server for the health check.
+// It returns an error if the setup failed, or runs the server in a goroutine.
+// Stop the server by cancelling the passed context.
+func Serve(ctx context.Context, port int) error {
+	if port == 0 {
+		return errors.New("port should be non-zero")
+	}
+	ln, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%v", port))
+	if err != nil {
+		return err
+	}
+
+	srv := &http.Server{
+		Handler:           healthHandler{},
+		ReadTimeout:       defaultTimeout,
+		ReadHeaderTimeout: defaultTimeout,
+		WriteTimeout:      defaultTimeout,
+	}
+
+	go srv.Serve(ln)
+	go closeOnContext(ctx, srv)
+	return nil
+}
+
+func closeOnContext(ctx context.Context, srv *http.Server) {
+	<-ctx.Done()
+	srv.Close() // srv will close the listener
+}
+
+type healthHandler struct{}
+
+func (h healthHandler) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+	health, err := health.GetStatusNonBlocking()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+	var b strings.Builder
+	b.WriteString("Agent health: ")
```

Comment:
WDYT about responding a json that might be easier to parse instead of raw text?

Reply:
Output is not printed in the `kubectl describe` command, I'm switching back to json.

As with the `agent health` command, a debug log is printed if a healthcheck shows unhealthy components. This should allow us to get more information from flares.